linode/manager
# M3-5809: Run Cypress via Docker Compose

## Description

**What does this PR do?**
This PR adds the capability to run Cypress via Docker, which will later be used to run Cypress in Jenkins.

Because this was built with Jenkins in mind, it is assumed that dependencies are already installed (for Linux systems) and that Cloud Manager is already built when the Docker container starts.

## How to test

**What are the steps to reproduce the issue or verify the changes?**

Before you can build and start the containers, you must install Node dependencies for Linux and build Cloud Manager. Because Docker volumes run slowly on MacOS, this will take noticeably longer than usual but thankfully only needs to be done once. Install Linux dependencies and build Cloud Manager using these commands:

```sh
# From `manager` repo root
# Run `yarn clean` to delete Node dependencies for MacOS
yarn clean

# Run `yarn install:all` from inside the web container to install Node dependencies for Linux
docker compose run web yarn install:all

# Run `yarn build` from inside the web container to build Cloud Manager using Linux dependencies
docker compose run web yarn build
```
Running the 3 commands above took 25 minutes on my machine.

Once your environment is set up, you can run Cypress via Docker Compose:

```sh
docker compose build
docker compose up e2e
```

Confirm that Cypress runs and that tests are generally passing.
